pdns/test-signers.cc

```diff
@@ -149,7 +149,10 @@ static const std::array<struct SignerParams, 3> signers
       DNSSECKeeper::ED25519,
       true,
 
-      std::nullopt},
+      std::make_optional(std::string{
```

Comment:
Perhaps just `#if defined(HAVE_LIBCRYPTO_ED25519)` as the sodium signer uses openssl for the export?

(Now I wonder if the PEM functions in `sodiumsigners.cc` should be wrapped in `ifdef` too)

Reply:
> Perhaps just `#if defined(HAVE_LIBCRYPTO_ED25519)` as the sodium signer uses openssl for the export?

The `PEM_*` functions are in `libcrypto`, so I think the accurate guard would be something like `(libsodium && libcrypto) || libcrypto_ed25519`. `libcrypto_ed25519` requires `libcrypto`: `AC_REQUIRE([PDNS_CHECK_LIBCRYPTO])` in `pdns_check_libcrypto_eddsa.m4`.

> (Now I wonder if the PEM functions in sodiumsigners.cc should be wrapped in ifdef too)

Do I get it correctly that we’re able to build without `libssl` (which requires `libcrypto`) but we can’t build without `libcrypto`? If that's correct, there wouldn’t be any need to guard anything in `sodiumsigners`.

Do I understand all of this correctly?

Reply:
> If that's correct, there wouldn’t be any need to guard anything in sodiumsigners.

I wonder if there can be a situation where we have sodium, and libcrypto, but libcrypto does not support ed25519. In that case, sodium PEM handling would need a `crypto_ed25519` guard.

Otherwise, I think you're making sense :-)

Reply:
Right. Then it's the same for the tests: only a `crypto_ed25519` guard is needed. Because either we have `crypto_ed25519` or `libsodium` but if `libsodium` PEM exists, it means that `crypto_ed25519` exists.